BakeSkinning: add support for skinning normals with faceVarying interpolation

cameronwhite opened the pull request:
### Description of Change(s)
Previously, only normals with vertex or varying interpolation were deformed. This was something we noticed while experimenting with using UsdSkel to deform fractured meshes from rigid body simulations, where faceted normals are quite common.

To reproduce, load the [attached file](https://github.com/PixarAnimationStudios/USD/files/7605237/facevarying_normals.zip) with `usdview facevarying_normals.usda --camera /cameras/camera1`, and then run `UsdSkel.BakeSkinning(usdviewApi.stage.Traverse())` in the interpreter. The undeformed normals can be seen as the cube spins around.

A few notes:

- I debated between adding a separate `UsdSkelSkinFaceVaryingNormalsLBS()` method rather than more overloads of `UsdSkelSkinNormalsLBS()` - feedback on this is welcome!
- faceVarying normals also are not deformed in hydra, but the issue is less noticeable depending on the renderer (the normals are suppressed, leaving them to be auto-computed post skinning). I'd like to look at this for a future PR, but wanted to keep this one at a reasonable scope
- faceVarying normals are also not handled for blendshapes. Doing this would require extensions to the schema, which assumes a 1:1 mapping between normals and points.

cameronwhite committed:
BakeSkinning: support linear blend skinning for normals with faceVarying interpolation.

jilliene commented:
Filed as internal issue #USD-7043

ucbearkat commented:
@cameronwhite Thanks for the change! To answer your first question - using a separate UsdSkelSkinFaceVaryingNormalsLBS() method instead of adding yet another override is probably ok in this case since this is about the caller facing api and internally they all end up calling _SkinNormalsLBS anyway. 
Your caveats about the scope of the change are also noted. 
I've staged this internally for review and will get back to you with any comments that come up.

ucbearkat commented on pxr/usd/usdSkel/utils.cpp:
```diff
@@ -1236,7 +1236,34 @@ struct _InterleavedInfluencesFn {
+
+    size_t GetPointIndex(size_t index) const
+    {
+        int pointIndex = faceVertexIndices[index];
```
Can this be changed to const int?

ucbearkat commented on pxr/usd/usdSkel/utils.h:
```diff
@@ -595,6 +596,18 @@ UsdSkelSkinNormalsLBS(const GfMatrix3f& geomBindTransform,
+                                 TfSpan<const int> faceVertexIndices,
+                                 TfSpan<GfVec3f> normals,
+                                 bool inSerial=false);
+
```
For completeness and consistency with the other UsdSkelSkin* utility methods here, would it be possible to also provide an overload that takes GfMatrix3f transforms?

ucbearkat reviewed (commented):
@cameronwhite Sorry it took a while to get back to you on this. 
This change is pretty great overall. Noted a couple comments that came up.
Thanks!

cameronwhite committed:
BakeSkinning: address review feedback for UsdSkelSkinFaceVaryingNormalsLBS()

- Add a GfMatrix3f overload for UsdSkelSkinFaceVaryingNormalsLBS(),
  refactoring the implementation into a templated function.

- Use const where possible in _FaceVaryingPointIndexFn::GetPointIndex()

cameronwhite commented:
Thanks! I've pushed a commit with the noted changes
